### PR TITLE
Create threads named after its executor/event-loop

### PR DESCRIPTION
**Pull request by graemerocher**

### CODE REVIEW

**Comment by dstepanov** [comment]
There can be one problem if you have event loops and executors named the same, that’s why I suggested to be able to prefix or suffix the name of a created bean using EachBean

**Comment by graemerocher** [comment]
Hmm yeah that is not possibly right now. Would require a further change. @jameskleeh thoughts?

File: runtime/src/main/java/io/micronaut/scheduling/executor/UserExecutorConfiguration.java [hunk 1]
```diff
@@ -37,18 +41,53 @@
...
+     * @param name the name
+     * @param nThreads number of threads
+     * @param type the type
+     * @param parallelism the parallelism
+     * @param corePoolSize the core pool size
+     * @param threadFactoryClass the thread factory class
      */
-    protected UserExecutorConfiguration() {
+    @Inject
+    protected UserExecutorConfiguration(@Nullable @Parameter String name,
```
Comment (jameskleeh) [inline runtime/src/main/java/io/micronaut/scheduling/executor/UserExecutorConfiguration.java:73]: Are you certain this works? I believe this has to be annotated with `@ConfigurationInject`?

**Comment by jameskleeh** [comment]
Overall this seems like a pretty big change. This wouldn't go into a point release, perhaps 2.1. I don't know if there is any code out there that is relying on the existing names of the threads, so I'm not sure if this would be considered a breaking change or not. In general I'm in favor of having the thread names be more informative

**Comment by dstepanov** [comment]
The conflict can occur only if somebody managed to set up a custom event loop in 2.0 with the same name as an executor. Probably not a big probability.

**Comment by graemerocher** [comment]
I agree it is probably a good improvement for 2.1

**Comment by graemerocher** [comment]
@dstepanov can you rebase this onto 2.1.x

**Comment by dstepanov** [comment]
@graemerocher I don't think I can technically rebase your branch 😄 I would need to create a new PR

**Comment by graemerocher** [comment]
ah damn, forgot that this was mine.. will handle it. Sorry for the noise 🤦‍♂️

**Comment by graemerocher** [comment]
Thanks for the contribution!

**Pull request merged**